compiler/rustc_resolve/src/lib.rs

```diff
@@ -1686,21 +1686,43 @@ impl<'a> Resolver<'a> {
             .or_insert_with(|| self.arenas.alloc_name_resolution())
     }
 
+    // Test if AmbiguityError ambi is any identical to any one inside ambiguity_errors
+    fn is_identical_error(&mut self, ambi: &mut AmbiguityError<'_>) -> bool {
+        for ambiguity_error in &self.ambiguity_errors {
+            // if the span location and ident as well as its span are the same
+            if ambiguity_error.kind == ambi.kind
```

Comment:
Would it be better to add `#[derive(Eq)]` to `AmbiguityError` and then do `ambi == ambiguity_error` here?

Reply:
This isn't the same as `derive(Eq)` since it's (presumably) checking only the things that end up rendering in the error. I do think this should be pulled out into a method on `AmbiguityError` though.

This method should then also be renamed to something like `matches_previous_ambiguity_error` or something.

Reply:
I think Michael was right.  It is not a fully equal `#[derive(Eq)]`, because it doesn't check each field of the struct and its sub-field. It's a partial equal at most. And `matches_previous_ambiguity_error ` would be a better name as well.